webdriver-extension/index.bs

```diff
@@ -0,0 +1,535 @@
+The <dfn for="MockSensorType" enum-value>"relative-orientation"</dfn> type is the mock sensor type
+associated with the usage of the <a>RelativeOrientationSensor</a> interface.
+<dl>
+  <dt><a>Mock Sensor Reading Set</a></dt>
+  <dd>
+    <pre class="idl">
+      dictionary RelativeOrientationReadingSet : AbsoluteOrientationReadingSet {
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Geolocation Sensor ### {#mock-geolocation-sensor}
+The <dfn for="MockSensorType" enum-value>"geolocation"</dfn> type is the mock sensor type
+associated with the usage of the <a>GeolocationSensor</a> interface.
+<dl>
+  <dt><a>Mock Sensor Reading Set</a></dt>
+  <dd>
+    <pre class="idl">
+      dictionary GeolocationReadingSet {
+        required double? latitude;
+        required double? longitude;
+        required double? altitude;
+        required double? accuracy;
+        required double? altitudeAccuracy;
+        required double? heading;
+        required double? speed;
+      };
+    </pre>
+  </dd>
+</dl>
+
+### Proximity Sensor ### {#mock-proximity-sensor}
+The <dfn for="MockSensorType" enum-value>"proximity"</dfn> type is the mock sensor type
+associated with the usage of the <a>ProximitySensor</a> interface.
+<dl>
+  <dt><a>Mock Sensor Reading Set</a></dt>
+  <dd>
+    <pre class="idl">
+      dictionary ProximityReadingSet {
+        required double? distance;
+        required double? max;
+        required boolean? near;
+      };
+    </pre>
+  </dd>
+</dl>
+
+Extensions {#webdriver-extensions}
+=====================
+
+## Commands ## {#mock-sensor-commands}
+
+### Create mock sensor ### {#create-mock-sensor-command}
+
+<table>
+  <tbody>
+    <tr>
+      <th>HTTP Method</th>
+      <th><a lt="extension command uri template">URI Template</a></th>
+    </tr>
+    <tr>
+      <td>POST</td>
+      <td>/session/{session id}/sensor</td>
+    </tr>
+  </tbody>
+</table>
+
+The <dfn>Create mock sensor</dfn> <a>extension command</a> simulates user
+creation of a <a>mock sensor</a>.
+
+The <a>remote end steps</a> are:
+1.  Let |configuration| be the |configuration| argument, <a>converted to an IDL value</a>
+    of type {{MockSensorConfiguration}}. If this throws an exception, return a
+    <a>WebDriver error</a> with <a>WebDriver error code</a> <a>invalid argument</a>.
+2.  Let |type| be the |configuration|.{{MockSensorConfiguration/mockSensorType}}, if <a>current browsing context</a>
+    already exists this |type| of [=mock sensor=], return a <a>WebDriver error</a> with <a>WebDriver error code</a>
+    [=mock sensor already created=].
+3.  Run these sub-steps [=in parallel=] to create a [=mock sensor=] in <a>current browsing context</a>:
+  1.  Set the [=mock sensor type|mock sensor's type=] to |type|.
+  2.  Let |shouldConnected| be the |configuration|.{{MockSensorConfiguration/sensorShouldConnected}}, if |shouldConnected| is true,
+	  1.  User agent should fail to [=connect to mock sensor=].
```

Comment:
What if it succeeds?

Reply:
I made a mistake on this.
`sensorShouldConnected` is a flag, when set to false, it should force the result of invoking [connect to sensor](https://w3c.github.io/sensors/#connect-to-sensor) to be false, otherwise true.
Thus, I think we can do nothing if this operation fails. Right?